### PR TITLE
use official openjdk image

### PR DESCRIPTION
**Pull request by riuvshin**
### What does this PR do?
use official openjdk image which will allow us to fix jdk version and avoid unexpected updates / upgrades

#### Changelog
Use the official OpenJDK image.

### CODE REVIEW

**Review by benoitf — DISMISSED** [review]

**Comment by benoitf** [comment]
@riuvshin do we need jdk and not the jre ?

**Comment by riuvshin** [comment]
@benoitf we discussed that and we agreed to use jdk this may help us in some cases

**Comment by benoitf** [comment]
@riuvshin what are these cases ?

**Comment by riuvshin** [comment]
@benoitf main point was that we didn't tested running on jre that much as we always used jdk,
do you think jre is enough ? and we don't need jdk to run che or codenvy? maybe some java opts that we pass or debug scenarios may require jdk?

**Comment by benoitf** [comment]
@riuvshin I still don't see what should these java opts and debug scenario.
The user doesn't connect to che master from IDE terminal. And the tools can be as well installed with apk at runtime if required for a one time debugging.
but using JRE saves extra  10MB

**Comment by riuvshin** [comment]
yes we know that switch to jre will save few mb for us but for now the problem that there is no one who can say that we don;t need jdk here we just don;t know is this required or not, so for now we want change only official image

**Comment by riuvshin** [comment]
@benoitf @skabashnyuk updated to use `8u111-jre-alpine` tag

File: dockerfiles/che/Dockerfile [hunk 1]
```diff
@@ -20,18 +20,15 @@
...
-    JAVA_HOME=/usr/lib/jvm/default-jvm/jre \
-    PATH=${PATH}:${JAVA_HOME}/bin \
     DOCKER_VERSION=1.6.0 \
     DOCKER_BUCKET=get.docker.com \
     CHE_IN_CONTAINER=true
 
 RUN echo "http://dl-4.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories && \
-    apk upgrade --update && \
-    apk add --update ca-certificates curl openssl openjdk8 sudo bash && \
+    apk add --update curl openssl sudo bash && \
```
Comment (benoitf) [inline dockerfiles/che/Dockerfile:31]: maybe we could drop as well curl and openssl after docker is downloaded (but it's not related to this PR)

**Review by benoitf — APPROVED** [review]

File: dockerfiles/che/Dockerfile (hunk 1 above)
Comment (riuvshin) [reply dockerfiles/che/Dockerfile:31]: im not sure why we added this here so it require another discussion

**Review by skabashnyuk — APPROVED** [review]

**Review by bmicklea — APPROVED** [review]

**Pull request merged**